packages/webdriverio/src/commands/browser/switchWindow.ts

```diff
@@ -64,6 +64,14 @@ export default async function switchWindow (
         if (matchesTarget(title)) {
             return tab
         }
+
+        /**
+         * check window name
+         */
+        const windowName:string = await this.execute('return typeof(window) !== "undefined" && window.name')
+        if (Object.keys(windowName).length && matchesTarget(windowName)) {
```

Comment:
Add the length check to the `matchesTarget` method for readability and abstraction